File: packages/tmdb_repository/lib/src/models/search_result.dart
import 'package:equatable/equatable.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:tmdb_api/tmdb_api.dart';

part 'search_result.g.dart';

@JsonSerializable()
class SearchResult extends Equatable {
  const SearchResult({
    required this.id,
    required this.title,
  });

  final int id;
  final String title;

  factory SearchResult.fromJson(Map<String, dynamic> json) =>
      _$SearchResultFromJson(json);

  Map<String, dynamic> toJson() => _$SearchResultToJson(this);

  factory SearchResult.fromMovieRecord(MovieRecord movieData) {
    return SearchResult(id: movieData.id, title: movieData.title);
  }

  @override
  List<Object> get props => [
        id,
        title,
      ];
}

File: server/cinemadle_datamodel/search_result.go
package cinemadle_datamodel

import "encoding/json"

type SearchResult struct {
	Id    int    `json:"id"`
	Title string `json:"title"`

	AdditionalInformation map[string]string `json:"additionalInformation"`
}

func (m SearchResult) ToJson() (string, error) {
	j, err := json.Marshal(m)

	if err != nil {
		return "", err
	}

	return string(j), nil
}
